common/flatpak-dir.c

```diff
@@ -4829,7 +4829,7 @@ flatpak_dir_pull (FlatpakDir                           *self,
       else
         {
           flatpak_remote_state_lookup_ref (state, ref, &rev, NULL, error);
-          if (rev == NULL && error != NULL && *error == NULL)
```

Comment:
Might this result in overwriting `error`?

Reply:
All the code above this does `goto out;` when it sets `error` so the only way we could be overwriting `error` is if there's a bug somewhere with respect to the rules of using `GError`, and I don't think we need to account for that case. (If we did, we would need a lot more `if (*error == NULL)` checks all over the codebase, I think.)

Reply:
As in, `flatpak_remote_state_lookup_ref()` presumably sets `error` iff it returns `(rev == NULL)`. In that case, the branch below immediately calls `g_set_error (error, …)` which would result in overwriting `error`.

Reply:
You're right. I managed to not see that despite staring right at it

Reply:
I know that feeling :-)